cmd/agent/subcommands/analyzelogs/command.go

```diff
@@ -108,34 +135,78 @@ func runAnalyzeLogs(cliParams *CliParams, config config.Component) error {
 }
 
 // Used to make testing easier
-func runAnalyzeLogsHelper(cliParams *CliParams, config config.Component) (chan *message.Message, *launchers.Launchers, pipeline.Provider) {
+func runAnalyzeLogsHelper(cliParams *CliParams, config config.Component, ac autodiscovery.Component) (chan *message.Message, *launchers.Launchers, pipeline.Provider, error) {
 	configSource := sources.NewConfigSources()
+	sources, err := getSources(ac, cliParams)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	for _, source := range sources {
+		if source.Config.TailingMode == "" {
+			source.Config.TailingMode = "beginning"
+		}
+		configSource.AddSource(source)
+	}
+	return agentimpl.SetUpLaunchers(config, configSource)
+}
+
+func getSources(ac autodiscovery.Component, cliParams *CliParams) ([]*sources.LogSource, error) {
+	sources, err := resolveFileConfig(cliParams)
+	if err == nil {
+		return sources, nil
+	}
+
+	sources, err = resolveCheckConfig(ac, cliParams)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid file config path or check name")
```

Comment:
Does it make sense to nest the underlying `err` (from line `160`)?
(If that error doesn't provide any actionable context, then you can ignore this comment)